GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -229,6 +229,18 @@ public virtual void Shutdown()
             this.indexParsingThread.Wait();
         }
 
+        public void UpdateProjection()
+        {
+            // Inform the index parsing thread that the index has changed,
+            // which means the projection needs to be updated to be correct before git proceeds
+            this.parseIndexRequested.Set();

```

Comment:
I've read through this a few times now, and I'm not sure that we need `parseIndexRequested`.  Previously we used `externalLockReleaseRequested` to force the parsing thread to wait for the git command to finish before it runs, but it seems that now we always want it to run right away.

Did you have any scenarios in mind that where `parseIndexRequested` is useful?

Reply:
I don't _think_ we need this anymore.  The code that runs in between waking up the thread and releasing it here is the status cache invalidation and newly created list clearing which I don't think would affect the projection getting updated. I will remove this and run the tests to make sure.

Reply:
> The code that runs in between waking up the thread and releasing it here is the status cache invalidation and newly created list clearing which I don't think would affect the projection getting updated. I will remove this and run the tests to make sure.

I have not been able to come up with any scenarios where that would be an issue.  Because we're in a git command we would have already finished processing anything that uses the list before the command starts (I think).  

I was thinking if this causes an issue we could move the clearing of the list up (so that it happens before starting the index parse), but I'm not sure if there are any scenarios where that would be an issue.  I think it comes down to whether we need to guarantee that the list has been cleared when we release the index-changed hook.